activerecord/test/activejob/destroy_association_async_job_test.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+
+class UndefinedConstantAsync < ActiveRecord::Base
+  self.destroy_association_async_job = "UndefinedConstantJob"
+end
+
+autoload :UnloadableBaseJob, "activejob/unloadable_base_job"
+class UnloadableBaseAsync < ActiveRecord::Base
+  self.destroy_association_async_job = "UnloadableBaseJob"
+end
+
+class UnusedBelongsToAsync < ActiveRecord::Base
+  self.destroy_association_async_job = nil
+end
+
+class UnusedHasOneAsync < ActiveRecord::Base
+  self.destroy_association_async_job = nil
+end
+
+class UnusedHasManyAsync < ActiveRecord::Base
+  self.destroy_association_async_job = nil
+end
+
+
+class DestroyAssociationAsyncJobTest < ActiveRecord::TestCase
+  test "destroy_association_async_job requires valid job class" do
+    error = assert_raises NameError do
+      UndefinedConstantAsync.belongs_to :unused_destroy_asyncs, dependent: :destroy_async
+    end
+    assert_match %r/destroy_association_async_job: uninitialized constant UndefinedConstantJob/, error.message
+  end
+
+  test "destroy_association_async_job error shows a missing parent job class, as if ActiveJob were missing" do
+    error = assert_raises NameError do
+      UnloadableBaseAsync.belongs_to :unused_belongs_to, dependent: :destroy_async
+    end
+    assert_match %r/destroy_association_async_job: uninitialized constant PretendActiveJobIsNotPresent/, error.message
+  end
+
+  test "deprecation of rescuing ActiveJobRequiredError which has been replaced by a NameError" do
+    deprecation_message = capture(:stderr) do
+      UnloadableBaseAsync.belongs_to :unused_belongs_to, dependent: :destroy_async
+    rescue ActiveRecord::ActiveJobRequiredError => e
+      assert_kind_of NameError, e
+    end
+    assert_match %r/Use NameError instead/, deprecation_message
```

Comment:
To test that we show a deprecation warning regardless of whether the `rescue` is evaluated, this should be:

```suggestion
    assert_deprecated { ActiveRecord::ActiveJobRequiredError }
```